packages/itwin/reports-config-widget/src/widget/components/Reports.tsx

```diff
@@ -107,6 +115,49 @@ export const Reports = () => {
     [reports, searchValue]
   );
 
+  useEffect(() => {
+    if (jobStarted) {
+      const interval = window.setInterval(async () => {
+        const states = await bulkExtractor.getStates(reports.map((r) => r.id));
+        if (Array.from(states.values())
+          .filter((x) => x === ExtractionStates.Succeeded ||
+            x === ExtractionStates.Failed ||
+            x === ExtractionStates.None)
+          .length === states.size) {
+          setJobStarted(false);
+        }
+        setExtractionStates(states);
+      }, STATUS_CHECK_INTERVAL);
+      return () => window.clearInterval(interval);
+    }
+    return;
+  }, [bulkExtractor, reports, jobStarted]);
+
+  useEffect(() => {
+    if (!jobStarted) {
+      const timeout = window.setTimeout(() => {
+        setExtractionStates(new Map<string, ExtractionStates>());
+      }, STATUS_CHECK_INTERVAL);
+      return () => window.clearTimeout(timeout);
+    }
+    return;
+  }, [jobStarted]);
+
+  function onClickTile(e: any, report: Report) {
```

Comment:
Make this into an arrow function.

Reply:
function made into arrow function. Is `useCallback` really necessary here?

Reply:
Not necessary, disregard that. Nothing is depending on that function.